### ISSUE TITLE
Using secrets for Docker build unable to get pip.conf

### ISSUE THREAD

**Issue by rajendar38** [GoogleContainerTools/kaniko#1850]
Not Reading  secret from /kaniko folder .
Explain: in ADO pipeline we are creating kaniko folder and placing in pip.conf in that folder 

**Expected behavior**
should read pip.conf from kaniko folder 
**To Reproduce**
      Docker.env file contents 
      #Get the environment
      ENV PIP_CONFIG_FILE /kaniko/pip.conf
      #COPY /kaniko/pip.conf /etc/pip.conf
      RUN  conda env update --name base --file /environment.yml --prune && conda clean -afy_

if we do a copy we are able to do conda update , but if I use pip ENV variable we are unable to update it .

**Comment by LaurentTrk** [comment]
Hi @rajendar38,

If you use the `COPY /kaniko/pip.conf /etc/pip.conf` statement, it works right ?

Could you try with this statement ?
```dockerfile
RUN PIP_CONFIG_FILE=/kaniko/pip.conf conda env update --name base --file /environment.yml --prune && conda clean -afy
```

**Comment by rajendar38** [comment]
> Hi @rajendar38,
> 
> If you use the `COPY /kaniko/pip.conf /etc/pip.conf` statement, it works right ?
> 
> Could you try with this statement ?
> 
> ```dockerfile
> RUN PIP_CONFIG_FILE=/kaniko/pip.conf conda env update --name base --file /environment.yml --prune && conda clean -afy
> ```

Yes we have tried it but it is unable find location /kaniko/pip.config as if we are not explicitly copying.
One more thing I forgot to mention, we are using blob storage as our context , (https://)
But not [dir:///] , 
We are running ado pipeline we want to run build from within the cluster.
We don’t cluster node access so we are using blob storage  as  our storage

**Comment by LaurentTrk** [comment]
One last check:
```dockerfile
RUN cat /kaniko/pip.conf
```

Does this statement print the contents of your pip configuration file ?

**Comment by rajendar38** [comment]
> ```dockerfile
> RUN cat /kaniko/pip.conf
> ```

@LaurentTrk 
Now my Docker file is something like this .
ENV PIP_CONFIG_FILE /kaniko/pip.conf
#COPY /kaniko/pip.conf /etc/pip.conf
RUN cat /kaniko/pip.conf
RUN  PIP_CONFIG_FILE=/kaniko/pip.conf conda env update --name base --file /environment.yml --prune  && conda clean -afy
**Note** : I am using https: // as my prefix in the context not dir:///

**Comment by LaurentTrk** [comment]
Do you see the contents of the pip.conf in the logs of the building step ?

**Comment by rajendar38** [comment]
> 

The only difference I see from #489 issue was you explained scenario with dir:/// context here this context https://
doent it make any difference ??

**Comment by LaurentTrk** [comment]
> > 
> 
> The only difference I see from #489 issue was you explained scenario with dir:/// context here this context https:// doent it make any difference ??

I don't know the differences...

Where is your  `pip.conf` file generated ? In the context archive ? In the building container ?

**Comment by rajendar38** [comment]
in the context archive

**Comment by LaurentTrk** [comment]
Humm, that could be the issue. 

In my case, the `/kaniko` folder is mounted and accessible during the build, but resides outside of the context. It is literally the `/kaniko` folder of the building container.

**Comment by rajendar38** [comment]
> Humm, that could be the issue.
> 
> In my case, the `/kaniko` folder is mounted and accessible during the build, but resides outside of the context. It is literally the `/kaniko` folder of the building container.

Means 
you have PV and PVC  created for storage class: local storage attached to kaniko pod  right ?
if so 
is this  `kaniko` folder is in the k8s cluster node . ?
how actually design done ?

**Comment by LaurentTrk** [comment]
No PV nor PVC involved, just a regular local folder. The  `pip.conf` is created in the building container as part of the build, before calling the  `kaniko` command.

**Comment by rajendar38** [comment]
> No PV nor PVC involved, just a regular local folder. The `pip.conf` is created in the building container as part of the build, before calling the `kaniko` command.

I didnot undersatand that , can you please eleborate 
what is meant by building container?
how we will get control on build container without been building it 
what is meanst by kaniko command ?
Here I am running kaniko as pod inside k8s cluster in my scenario
invoking with kubectl create command

**Comment by LaurentTrk** [comment]
I am using it in a Gitlab pipeline, so that's a very different usage.

In your case, I think the use of a `ConfigMap` should do the trick. Something like:
```bash
kubectl create configmap pip-config --from-file=pip.conf
```

Then in your deployment :
```yaml
...
  containers:
    - name: kaniko
      ...
      volumeMounts:
      - name: pip-config-volume
        mountPath: /kaniko
  volumes:
    - name: pip-config-volume
      configMap:
        name: pip-config
...
```

_(not sure of the whole syntax, but you get the idea :) )_

**Comment by rajendar38** [comment]
> ```dockerfile
> IP_CONFIG_FILE=/kaniko/pip.conf 
> ```

I have tried end up in different issue 
  Normal   Created    2m34s  kubelet            Created container kaniko
  Warning  Failed     2m34s  kubelet            Error: failed to create containerd task: OCI runtime create failed: container_linux.go:380: starting container process caused: exec: "/kaniko/executor": stat /kaniko/executor: no such file or directory: unknown

**Comment by LaurentTrk** [comment]
Seems that mounting the whole  `kaniko` folder is not a good idea, you could try with a subfolder (let's say `/kaniko/pip`) and set it accordingly in the `dockerfile` (`PIP_CONFIG_FILE /kaniko/pip/pip.conf`).

**Comment by rajendar38** [comment]
> /kaniko/pip

Thank you . you saved my day , how did you figured it out .
now it is working fine , Thanks a lot
